compiler/noirc_evaluator/src/ssa/opt/mem2reg/alias_set.rs

```diff
@@ -9,71 +9,116 @@ use crate::ssa::ir::value::ValueId;
 /// Note that we distinguish between "definitely has no aliases" - `Some(BTreeSet::new())`, and
 /// "unknown which aliases this may refer to" - `None`.
 #[derive(Debug, Default, Clone)]
-pub(super) struct AliasSet {
-    aliases: Option<BTreeSet<ValueId>>,
+pub(super) enum AliasSet {
+    #[default]
+    Unknown,
+    Empty,
+    Known(ValueId),
+    KnownMultiple(BTreeSet<ValueId>),
 }
 
 impl AliasSet {
     pub(super) fn unknown() -> AliasSet {
-        Self { aliases: None }
+        AliasSet::Unknown
     }
 
     pub(super) fn known(value: ValueId) -> AliasSet {
-        let mut aliases = BTreeSet::new();
-        aliases.insert(value);
-        Self { aliases: Some(aliases) }
+        AliasSet::Known(value)
     }
 
     pub(super) fn known_multiple(values: BTreeSet<ValueId>) -> AliasSet {
-        Self { aliases: Some(values) }
+        AliasSet::KnownMultiple(values)
```

Comment:
You should test the length of this set first to see if it actually has multiple ids

Reply:
Good catch, done!